hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/javassist/AttributeTypeDescriptor.java

```diff
@@ -79,6 +83,34 @@ public String buildInLineDirtyCheckingBodyFragment(JavassistEnhancementContext c
 		return builder.toString();
 	}
 
+	private void fetchValueBeforeComparisonIfNeeded(
+			JavassistEnhancementContext context,
+			CtClass managedCtClass,
+			CtField currentValue,
+			StringBuilder builder) {
+		if ( PersistentAttributesHelper.isAssignable( managedCtClass, PersistentAttributeInterceptable.class.getName() )
+				&& context.isLazyLoadable( currentValue ) ) {
+			// Should fetch the value by calling the GET method (if not loaded yet) , in order to initialize the value before the comparison,
+			// otherwise the original value = null and the dirty checking is wrong
+			builder.append( String.format(
+					" if( $1 == null " +
+							" && %1$s != null " +
+							" &&  %1$s instanceof %2$s " +
+							" &&  !((%2$s) %1$s).isAttributeLoaded(\"%3$s\") ) ",
+					EnhancerConstants.INTERCEPTOR_FIELD_NAME,
+					LazyAttributeLoadingInterceptor.class.getName(),
+					currentValue.getName()
+							)
+			);
+			builder.append( String.format(
+					"  {  get%s();  }",
```

Comment:
I am not sure the getter logic is extremely robust. For instance, if the property is a boolean type, the getter might follow the pattern of 'is%s'. Not sure whether there is existing stuff to borrowed from, but I noticed there is something similiar in ByteBuddyEnhancementContext#resolveGetter(FieldDescription). HIH.

Reply:
I changed the code to generate the code to PersistentAttributesHelper#findGetterOrNull.